src/daemon/common.c

```diff
@@ -270,6 +270,8 @@ ssize_t swrite (int fd, const void *buf, size_t count)
 	ssize_t     status;
 	struct      pollfd pfd;
 
+	assert (fd >= 0);
```

Comment:
I was thinking calling swrite with a negative fd in the callback struct is a programming error.
Meanwhile it helped me catch the bug I was looking for.

I'll turn this into a negative return instead.